app/models/mixins/miq_web_server_runner_mixin.rb

```diff
@@ -14,4 +14,30 @@ def do_before_work_loop
     # Since puma/thin traps interrupts, log that we're going away and update our worker row
     at_exit { do_exit("Exit request received.") }
   end
+
+  module ClassMethods
+    def start_worker(*args)
+      runner = self.new(*args)
+      _log.info("URI: #{runner.worker.uri}")
+
+      # Do all the SQL worker preparation in the main thread
+      runner.prepare
+
+      # The heartbeating will be done in a separate thread
+      Thread.new { runner.run }
+
+      runner.worker.class.configure_secret_token
```

Comment:
@matthewd Moved the configuring of the secrete_token + reset the secrets if app already initialized here